lib/Command/UpsertProvider.php

```diff
@@ -0,0 +1,91 @@
+<?php
+
+declare(strict_types=1);
+/**
+ * @copyright Copyright (c) 2021 Bernd Rederlechner tsdicloud@github.com>
+ *
+ * @license GNU AGPL version 3 or any later version
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+namespace OCA\UserOIDC\Command;
+
+use \Symfony\Component\Console\Command\Command;
+
+use OCA\UserOIDC\Db\ProviderMapper;
+
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+
+class UpsertProvider extends Command {
+	private $providerMapper;
+
+	public function __construct(ProviderMapper $providerMapper) {
+		parent::__construct();
+		$this->providerMapper = $providerMapper;
+	}
+
+	protected function configure() {
+		$this
+			->setName('user_oidc:provider')
+			->setDescription('Create, show or update a OpenId connect provider connfig given the identifier of a provider')
+			->addArgument('providerid', InputOption::VALUE_REQUIRED, 'Administrative identifier name of the provider in the setup')
```

Comment:
The issue with using the provider identifier as of right now is that it is not enforced to be unique currently, so the frontend as well as the mapper allow creating two providers with the same one. I'll prepare a patch to add that separately, so we can get this merged and I can then extend it with the attribute mapping in that pull request.